bpf/lib/nat.h

```diff
@@ -701,9 +701,15 @@ snat_v4_needs_masquerade(struct __ctx_buff *ctx __maybe_unused,
 	remote_ep = lookup_ip4_remote_endpoint(tuple->daddr, 0);
 	if (remote_ep && identity_is_remote_node(remote_ep->sec_identity)) {
 		/* Don't masquerade in native-routing mode: */
-		if (!is_defined(TUNNEL_MODE))
+		if (!is_defined(TUNNEL_MODE)) {
+			#ifdef ENABLE_REMOTE_NODE_SNAT
+			{
+				target->addr = IPV4_MASQUERADE;
+				return NAT_NEEDED;
```

Comment:
Did you consider adding a test for this change? I'm a bit worried we regress without one. [A BPF unit test](https://github.com/cilium/cilium/tree/main/bpf/tests) would probably be good enough.

Reply:
Fair enough :slightly_smiling_face:

Reply:
cc @julianwiedmann